sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -254,21 +260,15 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
         shouldExit = true
         lock.withLock {
           // TODO https://github.com/mapbox/mapbox-maps-android/issues/607
-          if (mapboxRenderer.needDestroy || mapboxRenderer is MapboxTextureViewRenderer) {
-            mapboxRenderer.onSurfaceDestroyed()
-            releaseEgl()
-            surface?.release()
+          if (mapboxRenderer is MapboxTextureViewRenderer) {
```

Comment:
just a question
should we check `eglPrepared` like in `destroy` method?

Reply:
basically I'm using this flag to check if `releaseAll` was called, I think we actually need another flag for it as EGL could be released in some other cases as well and that could lead us to a trouble.

Reply:
after given another thought let's leave as is, in this particular case current way is the correct one. Adding new flag just brings more complexity, we already have a bunch of those.